File: erp_flutter/lib/core/models/report.dart
class Report {
  final int id;
  final String title;
  final String module;
  final String type;
  final String period;
  final String status;
  final String createdBy;
  final DateTime createdDate;
  final DateTime? dueDate;
  final int completionPercentage;

  Report({
    required this.id,
    required this.title,
    required this.module,
    required this.type,
    required this.period,
    required this.status,
    required this.createdBy,
    required this.createdDate,
    this.dueDate,
    required this.completionPercentage,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      module: json['module'] ?? '',
      type: json['type'] ?? '',
      period: json['period'] ?? '',
      status: json['status'] ?? '',
      createdBy: json['createdBy'] ?? '',
      createdDate: json['createdDate'] != null ? DateTime.parse(json['createdDate']) : DateTime.now(),
      dueDate: json['dueDate'] != null ? DateTime.parse(json['dueDate']) : null,
      completionPercentage: json['completionPercentage'] ?? 0,
    );
  }
}

File: erp_api/ERP.API/Features/Reports/Report.cs
namespace ERP.API.Features.Reports;

public class Report
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Module { get; set; } = string.Empty; // HR, Finance, Purchase, Assets, Services, Projects, Documents
    public string Type { get; set; } = "Report"; // Plan, Report
    public string Period { get; set; } = string.Empty; // Q1 2024, Annual 2024, etc.
    public string Status { get; set; } = "Draft"; // Draft, InProgress, Completed, Approved
    public string CreatedBy { get; set; } = string.Empty;
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public DateTime? DueDate { get; set; }
    public int CompletionPercentage { get; set; }
}
